### ISSUE TITLE
[autocomplete] accessibility: Limited results status message not announced by screen readers

### ISSUE THREAD

**Issue by grace-snow** [mui/base-ui#4181]
# Bug report

## Current behavior

In the [Limit Results demo of the autocomplete component](https://base-ui.com/react/components/autocomplete#limit-results), screen readers don't announce the message "Hiding # results (type a more specific query to narrow results" 

![The open autocomplete with message at the end of the listbox highlighted explaining that the number of visible suggestion is limited](https://github.com/user-attachments/assets/bfe3f3a2-3e3c-4405-97f7-3a28ac794847)

Tested with
* VoiceOver in Chrome and Safari on Mac OS 
* NVDA in Chrome on Windows
* JAWS in Chrome on Windows

## Expected behavior

All screen readers should announce the full message so that screen reader users know suggestions have been limited and how many are hidden.

This is currently an issue under [WCAG 2.2 Success Criterion 4.1.3 Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html) and possibly [WCAG 2.2 Success Criterion 3.3.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html)

## Reproducible example

Steps to reproduce:
1. Navigate to [the Autocomplete Limit Results demo](https://base-ui.com/react/components/autocomplete#limit-rresults)
2. Turn on your screen reader
3. Type a single letter, such as "a", into the input
4. Note that the message at the bottom of the listbox about hiding results is not announced

## Base UI version

v1.2.0

## Which browser are you using?

Safari and Chrome

## Which OS are you using?

Mac OS and Windows

## Which assistive tech are you using (if applicable)?

VoiceOver, NVDA and JAWS

## Additional context

Adding `aria-describedby` to the listbox referencing the unique `id` of the status message may fix the announcement.

There may be a problem with the way the status message has been coded, which is causing it to not be announced. Potential solutions may be: 
* to add a slight delay before populating the message inside the aria-live region
* using an assertive live announcement (or alert role) 
* removing aria-atomic

**Comment by atomiks** [comment]
> Adding aria-describedby to the listbox referencing the unique id of the status message may fix the announcement.

Will look into this idea.

> There may be a problem with the way the status message has been coded, which is causing it to not be announced. Potential solutions may be:

Yes the issue is aria-live regions don't announce their content when first mounted, they only announce content changes (so the delay idea would work but this is problematic when trying to avoid visible layout shifts, etc.)
